packages/components/src/autocomplete/index.tsx

```diff
@@ -440,9 +395,11 @@ export function useAutocompleteProps( options ) {
 
 	const mergedRefs = useMergeRefs( [
 		ref,
-		useRefEffect( ( element ) => {
-			function _onKeyDown( event ) {
-				onKeyDownRef.current( event );
+		useRefEffect( ( element: HTMLElement ) => {
+			function _onKeyDown( event: KeyboardEvent ) {
+				if ( typeof onKeyDownRef.current !== 'undefined' ) {
+					onKeyDownRef.current( event );
+				}
```

Comment:
Is this simpler with just an optional?

```suggestion
				onKeyDownRef.current?.( event );
```

Reply:
It looks like my original comment replying to this was lost somehow... but yes, that looks a lot simpler. It's also a cool use of an optional. I've used it for properties many times but I don't think I've encountered it on a n actual call like this before. I like it!